src/TabSwipe/widget/TabSwipe.ts

```diff
@@ -0,0 +1,221 @@
+            selectionClass: ".mx-tabcontainer-content",
+            swipeClass: "swipe-tab",
+            targetWidgetType: "mxui.widget.TabContainer"
+        };
+        this.targetNode = this.findTargetNode(this.targetName);
+        this.targetWidget = this.targetNode ? registry.byNode(this.targetNode) : null;
+        if (this.checkCompatibility(this.targetWidget)) {
+            this.targetWidget.connectedTabSwipe = this.id;
+            domClass.add(this.targetWidget.domNode, this.settings.swipeClass);
+        }
+        console.log("========== POSTCREATE==========");
+        this.targetWidget.getChildren().forEach( (child: TabPan) => {
+            if (child.index === 0) console.log("========");
+            console.log(`child ${child.index} width: ${child.domNode.offsetWidth}, height: ${child.domNode.offsetHeight} `);
+        });
+    }
+
+    /**
+     * called when context is changed or initialized.
+     * Though we are not interested in the context object it self, Tab could be updated on it.
+     * @param {type} obj - object in context
+     * @param {type} callback - callback when update is completed
+     * @returns {undefined}
+     */
+    update(contextObject: mendix.lib.MxObject, callback: () => void) {
+        if (this.targetWidget) {
+            console.log("============== UPDATE===========");
+            this.targetWidget.getChildren().forEach( (child: TabPan) => {
+                if (child.index === 0) console.log("========");
+                console.log(`child ${child.index} width: ${child.domNode.offsetWidth}, height: ${child.domNode.offsetHeight} `);
+            });
+            domClass.add(this.targetNode, this.settings.swipeClass);
+            // this.getVisableTabs();
+            this.setTabNavStyle(); // indicators
+            // this.setTabIndexZ(); // setting z-index
+            this.hammerTime();
+            this._setupEvents();
+
+            // this.loadAllTabContent(this.targetWidget, this.targetWidget._active);
+        }
+        callback();
+    }
+
+    /**
+     * change te tab container button, transform in to carousel indicators.
+     */
+    setTabNavStyle() {
+        domClass.remove(this.targetNode, "use-indicators");
+        if (this.tabNavStyle === "indicators") {
+            domClass.add(this.targetNode, "use-indicators");
+            const navNode = this.targetWidget._tabList;
+            domClass.remove(navNode, [ "nav", "nav-tabs", "mx-tabcontainer-tabs" ]);
+            domClass.add(navNode, "carousel-indicators");
+        }
+    }
+    setTabIndexZ() {
+        const tabChildren = this.targetWidget.getChildren() as TabPan[];
+        tabChildren.forEach(tabPan => tabPan.domNode.style.zIndex = `${tabChildren.length - tabPan.index}`);
+    }
+
+    /**
+     * connect target widget events
+     */
+    _setupEvents() {
+        // connect to tab clicking
+        this.own(aspect.after(this.targetWidget, "showTab", (deferred: any, args: any) => {
```

Comment:
Results in undefined, else I'll use javascript standard `arguments` as `arguments[1][0] to get the tab.